[JSC] Wasm IPInt: always parse SIMD

dhecht opened the pull request:
#### 171ff745282a48fbb645a9fd5754ffe12ad7798e
<pre>
[JSC] Wasm IPInt: always parse SIMD
<a href="https://bugs.webkit.org/show_bug.cgi?id=302402">https://bugs.webkit.org/show_bug.cgi?id=302402</a>
<a href="https://rdar.apple.com/164566289">rdar://164566289</a>

Reviewed by Yusuke Suzuki and Keith Miller.

Now that SIMD is fully supported in IPInt, there&apos;s no need to avoid
parsing SIMD instructions/types when parsing for IPInt. We can
always just generate the metadata even if we won&apos;t be using it
if --useWasmIPIntSIMD=false is set (which sticks around only until
relaxed SIMD is complete). This is a better trade off since it removes
a conditional branch from SIMD paths when parsing.

And since we always parse SIMD, we can get rid of the
assertAboutStackSize hack.

I&apos;ll follow this up with a cleanup change to remove
Context::tierSupportsSIMD() since all tiers now have this hardcoded
to true.

Testing: Wasm stress tests with --testing build
Canonical link: <a href="https://commits.webkit.org/302935@main">https://commits.webkit.org/302935@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/492a177da122b39636c5ac42aea9d09d9527f206

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130652 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2923 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/41607 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138073 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/82277 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/25a2cf70-39a0-4bce-859b-4f7822cec778) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2938 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2816 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/99538 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/67401 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/82d1c4e3-919e-48ad-a5a8-b99b879bf26d) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/133599 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2128 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116985 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/80246 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/5e8ff9ad-d4c0-4f76-9604-fe4aad272e9c) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2052 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35118 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/81329 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/122655 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110629 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35620 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/140553 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/129105 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/2713 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/2455 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108042 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2757 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113328 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107975 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27479 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2090 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31755 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/55702 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2783 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/66172 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/162120 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/2602 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/40431 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/2804 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/2709 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Source/JavaScriptCore/wasm/WasmIPIntGenerator.cpp:
```diff
@@ -527,14 +527,6 @@ class IPIntGenerator {
     PartialResult WARN_UNUSED_RETURN addUnreachable();
     PartialResult WARN_UNUSED_RETURN addCrash();
 
-    inline void assertAboutStackSize(bool condition)
```
Wow gross. Glad that's gone.

kmiller68 reviewed (approved):
r=me

Constellation reviewed (commented):
Maybe we should remove useWasmIPIntSIMD?

dhecht committed:
[JSC] Wasm IPInt: always parse SIMD
https://bugs.webkit.org/show_bug.cgi?id=302402
rdar://164566289

Reviewed by Yusuke Suzuki and Keith Miller.

Now that SIMD is fully supported in IPInt, there's no need to avoid
parsing SIMD instructions/types when parsing for IPInt. We can
always just generate the metadata even if we won't be using it
if --useWasmIPIntSIMD=false is set (which sticks around only until
relaxed SIMD is complete). This is a better trade off since it removes
a conditional branch from SIMD paths when parsing.

And since we always parse SIMD, we can get rid of the
assertAboutStackSize hack.

I'll follow this up with a cleanup change to remove
Context::tierSupportsSIMD() since all tiers now have this hardcoded
to true.

Testing: Wasm stress tests with --testing build
Canonical link: https://commits.webkit.org/302935@main
